src/rules/flake8_logging_format/rules.rs

```diff
@@ -0,0 +1,238 @@
+fn check_log_record_attr_clash(checker: &mut Checker, extra: &Keyword) {
+    match &extra.node.value.node {
+        ExprKind::Dict { keys, .. } => {
+            for key in keys {
+                if let Some(key) = &key {
+                    if let ExprKind::Constant {
+                        value: Constant::Str(string),
+                        ..
+                    } = &key.node
+                    {
+                        if RESERVER_ATTRS.contains(&string.as_str()) {
+                            checker.diagnostics.push(Diagnostic::new(
+                                LoggingExtraAttrClash(string.to_string()),
+                                Range::from_located(key),
+                            ));
+                        }
+                    }
+                }
+            }
+        }
+        ExprKind::Call { func, keywords, .. } => {
+            if let ExprKind::Name { id, .. } = &func.node {
+                if id == "dict" {
+                    for keyword in keywords {
+                        if let Some(key) = &keyword.node.arg {
+                            if RESERVER_ATTRS.contains(&key.as_str()) {
+                                checker.diagnostics.push(Diagnostic::new(
+                                    LoggingExtraAttrClash(key.to_string()),
+                                    Range::from_located(keyword),
+                                ));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        _ => {}
+    }
+}
+
+/// Check logging calls for violations
+pub fn logging_call(checker: &mut Checker, func: &Expr, args: &[Expr], keywords: &[Keyword]) {
+    if let ExprKind::Attribute { value, attr, .. } = &func.node {
+        if let ExprKind::Name { id, .. } = &value.node {
+            if ALLOWLIST.contains(&id.as_str()) {
+                return;
+            }
+
+            if let Some(logging_level) = LoggingLevel::from_str(attr.as_str()) {
+                let call_args = SimpleCallArgs::new(args, keywords);
+                let level_call_range = Range::new(
+                    Location::new(
+                        func.location.row(),
+                        value.end_location.unwrap().column() + 1,
+                    ),
+                    Location::new(
+                        func.end_location.unwrap().row(),
+                        func.end_location.unwrap().column(),
+                    ),
+                );
+
+                // G001 - G004
+                if let Some(format_arg) = call_args.get_argument("msg", Some(0)) {
+                    check_msg(checker, format_arg);
+                }
+
+                // G010
+                if checker.settings.rules.enabled(&Rule::LoggingWarn)
+                    && matches!(logging_level, LoggingLevel::Warn)
+                {
+                    let mut diagnostic = Diagnostic::new(LoggingWarn, level_call_range);
```

Comment:
One concern here is that `PGH002` does the same thing:

```
resources/test/fixtures/flake8_logging_format/G010.py:3:1: PGH002 `warn` is deprecated in favor of `warning`
```

We could consider deprecating that rule, though, in favor of this more comprehensive plugin. We'd basically just redirect `PGH002` to `G010`.

Reply:
I agree that it makes sense to prefer the logging plugin for this rule. Is it just a matter of adding it in `src/rule_redirects.rs`?

Reply:
Yeah, we'd need to delete all references to the rule (fixture files, etc.), then add the redirect to that map.